### ISSUE TITLE
Block import failed for #11543128 (0xf9bd…54b7)

### ISSUE THREAD

**Issue by jeremie-H** [openethereum/openethereum#204]
Hello i've got the following error in logs :  

```
Bad block detected: Error(Trie(InvalidStateRoot(0x21f0cc688883facb08a3e776b8d1be5f67907d1c9e33a6d2707b70349390669b)), State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })

...then after heapdump of RLP, tx. ...

Thread 'IO Worker #2' panicked at 'nonce will return Some when given BlockId::Latest. nonce was given BlockId::Latest. Therefore nonce has returned Some; qed', ethcore/src/client/traits.rs:93
```

- **OpenEthereum version (>=3.1.0)**: **v3.1.0-stable-2072341-20201102**
- **Operating system**: Linux / Ubuntu 20.04.1 LTS
- **Installation**: one-line installer
- **Fully synchronized**: yes, it was until this block
- **Network**: ethereum
- **Restarted**: yes

My openethereum node was really long to sync to head, but finally it **successfully** sync last week  :tada: (after 4 weeks to reach the last block in head). But now, i'm getting stuck with this particular block, with no reason. Even after restarting the node.  

my systemd exec line is the following : 
```
ExecStart=/home/eth/openethereum/openethereum --log-file=/home/eth/openethereum/out.log --base-path=/home/eth/openethereum/chain

```

i've got sufficient place on disk.  

In attachment, my logs from `journalctl --no-pager -u openethereum.service --since "15 minutes ago" --no-tail --quiet`  
[panic_qed_traitsrs.zip](https://github.com/openethereum/openethereum/files/5747805/panic_qed_traitsrs.zip)

**Comment by rakita** [comment]
Duplicate of #69 , it is fixed in `v3.1.1-rc.1` but you will need to resync again.

**Comment by jeremie-H** [comment]
Hello @rakita , when you said i need to resync, you don't want to say from the beginning ? 
i've spent 4 weeks to sync, i was thinking that openethereum was stable on 3.10 !

**Comment by jeremie-H** [comment]
is it possible to fix the database to work with the `v3.1.1-rc.1` ?

**Comment by i6mi6** [comment]
Just got the exact same problem with `v3.1.1-rc.1`
